play-routes-toolchain/transitions.bzl

```diff
@@ -0,0 +1,62 @@
+# This approach of incoming transition, store original values, outgoing
+# transition, reset to original values is inspired by what the rules_go
+# folks are doing.
+toolchain_setting_key = "@rules_play_routes_toolchain//:play-routes-toolchain"
+original_toolchain_setting_key = "@rules_play_routes_toolchain//:original-play-routes-toolchain"
+
+def _play_routes_toolchain_transition_impl(settings, attr):
+    """Update the rules_play_routes toolchain to the overridden value. Store
+    The original value, so it can be reset on the outgoing transition.
+    """
+    if attr.play_routes_toolchain_name == "":
+        return {}
+    elif settings[toolchain_setting_key] == attr.play_routes_toolchain_name:
+        # Do nothing when the attribute value is what the setting currently is
+        # otherwise needlessly change build settings by modifying the setting
+        # we store the original value in
+        return {}
+
+    new_settings = {}
```

Comment:
FWIW I like the way it is even if it is more verbose as it makes it clearer to those unfamiliar with toolchains that these are new settings because of the variable name.